cloud/cloud_file_system_impl.cc

```diff
@@ -786,19 +787,13 @@ IOStatus CloudFileSystemImpl::DeleteFile(const std::string& logical_fname,
   return st;
 }
 
-void CloudFileSystemImpl::RemoveFileFromDeletionQueue(
-    const std::string& filename) {
-  cloud_file_deletion_scheduler_->UnscheduleFileDeletion(filename);
-}
-
-void CloudFileSystemImpl::TEST_SetFileDeletionDelay(
-    std::chrono::seconds delay) {
-  cloud_file_deletion_scheduler_->TEST_SetFileDeletionDelay(delay);
-}
-
 IOStatus CloudFileSystemImpl::CopyLocalFileToDest(
     const std::string& local_name, const std::string& dest_name) {
-  RemoveFileFromDeletionQueue(basename(local_name));
+  if (cloud_file_deletion_scheduler_) {
+    // Remove file from deletion queue
+    cloud_file_deletion_scheduler_->UnscheduleFileDeletion(
```

Comment:
When will we write to files that have been moved to deletion queue? 

If this is still necessary, I'll update `DisaggStorageProvider` to cover this case (i.e., unscheduling file deletion in `PutCloudObject` call)